code/modules/mob/living/simple_animal/slime/life.dm

```diff
@@ -62,7 +62,7 @@
 				break
 
 			if(Target in view(1,src))
-				if(issilicon(Target))
+				if(issilicon(Target) || !CanFeedon(Target)) //If they're not able to be fed upon, ignore them.
```

Comment:
Make `issilicon` part of `CanFeedon`

Also you probably should make CanFeedon take an argument for being silent

Reply:
Will do!